fathom/src/core.rs

```diff
@@ -255,6 +261,207 @@ impl<'arena> Term<'arena> {
             }
         }
     }
+
+    // TODO: Add a new `Weaken` variant to `core::Term` instead of eagerly traversing the term?
```

Comment:
For the benefit of others, I believe Andras Kovacs’ staged language [has something similar](https://github.com/AndrasKovacs/staged/blob/9e381eb162f44912d70fb843c4ca6567b0d1683a/demo/Syntax.hs#L52)?